### PR TITLE
DPS device coverage

### PR DESCRIPTION
**Pull request by ngastelum-ms**

### CODE REVIEW

File: provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        private readonly string _globalDeviceEndpoint = "endpoint";
+        private readonly string _idScope = "id";
+
+        [TestMethod]
+        public void ProvisioningTransportRegisterRequest_DefaultPayload()
+        {
+            // arrange - act
+            var auth = new Mock<AuthenticationProvider>();
+            auth.Setup(p => p.GetRegistrationId()).Returns("registrationId");
```
Comment (abhipsaMisra) [inline provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs:28]: nit: save off "registrationId" into const?

File: provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+            var request = new ProvisioningTransportRegisterRequest(
+                _globalDeviceEndpoint,
+                _idScope,
+                auth.Object);
+
+            // assert
+            request.Payload.Should().BeNull();
```
Comment (abhipsaMisra) [inline provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs:36]: you could add asserts for regn Id, Id scope etc as well

**Review by abhipsaMisra — APPROVED** [review]

File: provisioning/device/tests/AuthenticationProviderX509Tests.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    public class AuthenticationProviderX509Tests
+    {
+        [TestMethod]
+        public void AuthenticationProviderX509_ThrowsWhenMissingCert()
+        {
+            Func<AuthenticationProviderX509> act = () => new AuthenticationProviderX509(null);
+            act.Should().Throw<ArgumentException>();
+        }
+    }
+}
```
Comment (brycewang-microsoft) [inline provisioning/device/tests/AuthenticationProviderX509Tests.cs:21]: extra line at the end of file

**Review by brycewang-microsoft — APPROVED** [review]

File: provisioning/device/tests/AuthenticationProviderX509Tests.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+namespace Microsoft.Azure.Devices.Provisioning.Client.UnitTests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class AuthenticationProviderX509Tests
+    {
+        [TestMethod]
+        public void AuthenticationProviderX509_ThrowsWhenMissingCert()
+        {
+            Func<AuthenticationProviderX509> act = () => new AuthenticationProviderX509(null);
```
Comment (schoims) [inline provisioning/device/tests/AuthenticationProviderX509Tests.cs:17]: // act
	            Func<AuthenticationProviderX509> act = () => new AuthenticationProviderX509(null);
	
	            // assert
	            act.Should().Throw<ArgumentException>();
	}

File: provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs (hunk 1 above)
Comment (schoims) [inline provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs:28]: is the setup necessary for this test?

File: provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        }
+
+        [TestMethod]
+        public void ProvisioningTransportRegisterRequest_WithPayload()
+        {
+            // arrange - act
+            var auth = new Mock<AuthenticationProvider>();
+            auth.Setup(p => p.GetRegistrationId()).Returns("registrationId");
+
```
Comment (schoims) [inline provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs:45]: same here, is this setup necessary?

File: provisioning/device/tests/AuthenticationProviderX509Tests.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            act.Should().Throw<ArgumentException>();
```
[the remark this answers is not in the document]
Comment (schoims) [reply provisioning/device/tests/AuthenticationProviderX509Tests.cs:18]: just curious why are we not throwing ArugmentNullException?

File: provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs (hunk 1 above)
Comment (ngastelum-ms) [reply provisioning/device/tests/ProvisioningTransportRegisterRequestTests.cs:28]: Good catch, I tested it without and looks like it's not needed.

**Pull request merged**